Fix rocket.surgery.nuke Update

JoeRobich opened the pull request:
Fix up #1336 

- Mimic the style of the RSG/Nuke repo's .build
- Regenerated GH actions
- Left in the tweaks around NuGet caching, the matrix of artifacts, and NuGet publishing

renovate[bot] committed:
chore(deps): update dependency rocket.surgery.nuke to v2

JoeRobich committed:
Fix up #1336

- Mimic the style of the RSG/Nuke repo's .build
- Regenerated GH actions
- Left in the tweak around NuGet caching, the matrix of artifacts, and NuGet publishing

JoeRobich committed:
Publish test results from MacOS run

JoeRobich commented:
Reviewing the CI logs it appears to be having trouble locating the test results and code coverage. Need to investigate further.

manandre commented:
I have found a possible explanation inside build logs:
```
[INF] "/home/runner/work/csharp-language-server-protocol/csharp-language-server-protocol/test/coverlet.runsettings" is not a valid Coverage settings file. Using default values.
```
But I do not see which part is considered as invalid.

manandre commented:
Solution found!
The DataCollector must be named `code coverage` and not `XPlat code coverage`.

JoeRobich committed:
Merge branch 'master' into dev/jorobich/fixup-1336

JoeRobich commented on test/coverlet.runsettings:
```diff
@@ -1,14 +1,116 @@
-</RunSettings>
+  <DataCollectionRunSettings>
+    <DataCollectors>
+      <DataCollector friendlyName="XPlat code coverage">
```
```suggestion
      <DataCollector friendlyName="code coverage">
```

JoeRobich committed:
Update test/coverlet.runsettings

manandre commented:
Surprising. The previous step is also using a relative path `/artifacts/logs/` and seems to work correctly.

JoeRobich committed:
Publish CI assets using absolute paths.

JoeRobich committed:
Merge branch 'master' into dev/jorobich/fixup-1336

JoeRobich committed:
Clear build warnings from sample/ and test/ folders.

Mainly acheived this by disabling nullable checking in files with nullability errors.

JoeRobich committed:
Try only generating Cobertura code coverage.

JoeRobich committed:
Merge branch 'master' into dev/jorobich/fixup-1336
